multi-node-testkit/src/main/scala/org/apache/pekko/remote/testconductor/Player.scala

```diff
@@ -215,8 +215,9 @@ private[pekko] class ClientFSM(name: RoleName, controllerAddr: InetSocketAddress
     case Event(msg: NetworkOp, _) =>
       log.error("received {} instead of Done", msg)
       goto(Failed)
-    case Event(_: ServerOp, _) =>
-      stay().replying(Status.Failure(new IllegalStateException("not connected yet")))
+    // Scala 3.3.1 compiler says this is unreachable
+    // case Event(_: ServerOp, _) =>
+    //  stay().replying(Status.Failure(new IllegalStateException("not connected yet")))
```

Comment:
I checked it and I think the compiler is right, reported it at https://github.com/lampepfl/dotty/issues/18519 too

Reply:
I think this line can be removed, the compiler is right.

Reply:
Do a swap
```scala
    case Event(msg: ServerOp, _) =>
      log.error("received {} instead of Done", msg)
      goto(Failed)
    case Event(_: NetworkOp, _) =>
      stay().replying(Status.Failure(new IllegalStateException("not connected yet")))
```